cstruct-async.opam

```diff
@@ -14,6 +14,7 @@ build: [
 ]
 depends: [
   "jbuilder" {build & >="1.0+beta7"}
+  "async"
```

Comment:
As far as I know, `Deferred.t` is not defined in `async_unix`. It is mentioned in the [interface](https://github.com/mirage/ocaml-cstruct/blob/master/async/async_cstruct.mli#L23).